src/parser/Clauses.h

```diff
@@ -285,9 +286,21 @@ class YieldColumn final {
 
   std::string toString() const;
 
+  bool isVisible() const {
+    return visible_;
+  }
+
+  void setVisible(bool visible) {
+    visible_ = visible;
+  }
+
  private:
   Expression *expr_{nullptr};
   std::string alias_;
+  // If some columns are required by order-by and are not explicitly
+  // declared in the return caluse, it is an invisible column and not to be
```

Comment:
IMHO, this is not a good fix.  Better to do that in optimizer,such as column pruning rule.

Reply:
I don't think so. It is not an optimization issue to me. It's a functional issue. Because this type of query shall be executable without the optimizer optimizing the plan.

Previously the validator reports column-not-found. Now it expands the scope for order-by when needed and possible by selecting required columns and adding them in the query plan. These columns are not to be expected in the result set. So I mark them to be invisible in their respective YieldColumn.

Reply:
You just need to make the filter execute ahead of time so that it can see all columns and the unexpected columns will not be projected in result set, which is also consistent with cypher's semantic definition of `with-where clause`:

![image](https://user-images.githubusercontent.com/26356194/205590846-1f3f9481-c7e9-497f-bfdd-2cff13b89732.png)

Reply:
Discussed offline. To summarize, we'd better prepare all data for potential aggregation and orderby before the final projection, instead of passing hidden column beneath the projection.